starcounter-include.html

```diff
@@ -476,13 +519,25 @@
          * @param {DocumentFragment} givenComposition
          */
         StarcounterIncludePrototype.stampComposition = function(givenComposition){
-            // create shadowRoot if none
             const shadowRoot = this.shadowRoot;
+            this.blockingShadowLinksLoaded && this.blockingShadowLinksLoaded.reject();
 
             if (givenComposition !== undefined) {
+                this.setAttribute('loading-presentation', '');
                 // reset SD to default CSS
                 shadowRoot.innerHTML = '';
 
+                // wait for blocking (not `[async]`) links
+                const blockingShadowLinksLoaded = blockingLinksLoaded(givenComposition);
+                this.blockingShadowLinksLoaded = blockingShadowLinksLoaded;
+
+                if(blockingShadowLinksLoaded){
+                    blockingShadowLinksLoaded.then(()=>{
+                        this.removeAttribute('loading-presentation');
+                    });
```

Comment:
Shouldn't you also add the function that removes the attribute in case of rejection?

```js
                    blockingShadowLinksLoaded.then(()=>{
                        this.removeAttribute('loading-presentation');
                    }, ()=>{
                        this.removeAttribute('loading-presentation');
                    });
```

Reply:
Please note, that `error` of `<link>` also resolves the promise.

The only rejection happens when the new composition is stamped (l.524). (or may happen if someday we will cal reject elsewhere)
This means that links for the given presentation were not loaded yet, meaning the content is still unstyled. Then why should we reveal it? The reason to reveal it will happen only once we load new composition with styled content.

Reply:
@tomalec explained to me that the only time when you will have the promise rejected is when a new view is being loaded in place of another view that was not loaded yet. So handling this case has no benefit, let's leave the `loading-presentation` attribute as is.